### PR TITLE
1180 auto assign annots to single encounter

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Allow form sightings to have multiple encounters (this should always have been the case and was a misunderstanding)
- If any submit has a single encounter, auto cluster the annots into the one encounter
- Quite a few test tweaks to allow this

DO NOT MERGE THIS UNTIL FE https://wildme.atlassian.net/browse/DEX-759 is merged

### CODE REVIEW

File: app/modules/asset_groups/metadata.py [hunk 1]
```diff
@@ -483,12 +483,6 @@ def process_request(self):
         self._validate_sightings()
         self.data_processed = AssetGroupMetadata.DataProcessed.sightings
 
-        if not self.bulk_upload and len(self.request['sightings'][0]['encounters']) != 1:
-            raise AssetGroupMetadataError(
-                log,
-                'Incorrect num encounters in form submission, there must be exactly one',
-            )
-
```
Comment (naknomum) [inline app/modules/asset_groups/metadata.py:491]: do we care that we lost the blocking of **zero** encounters here?  i know that _ultimately_ we must have > 1 encounter, but i am not sure if that means it cannot be empty _at this point_ (e.g. 1st encounter gets added later when it is a real sighting).

Comment (hwindsor) [reply app/modules/asset_groups/metadata.py:491]: I don't believe that we care. Also the FE will always send one so, no reason that we should

**Review by naknomum — APPROVED** [review]
lgtm

**Review by drewblount — APPROVED** [review]
Looks good t'me

**Pull request merged**